mitodl/mitxonline
# 1522: Remove instances of ErrorMessage for required fields

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1522

#### What's this PR do?
In order to improve accessibility, we have added the `required` attribute to required form fields.  Because of this, users are not able to submit forms if any of these `required` fields are blank.  This means that many of the formik `<ErrorMessage>` components never being accessible when the field's validation only checks that a value is defined for that field.

This PR also:
- Ensures forms are only validated when the user presses the submit button
- Adds more `autoComplete` where possible.

#### How should this be manually tested?
Visit the following forms within the application and verify that you cannot submit the form if any of the required fields are blank.  Verify that any validation error messages, unrelated to the field being required, are displayed (example: email format invalid).

Change email form:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with an empty email field.
- Ensure a validation tool-tip is shown when submitting the form with an invalid email format.  The form should not allow the user to submit.
- Ensure validation message is shown if the user uses their existing email for the email field upon submitting the form.
- Ensure you cannot submit the form with an empty password field.
- Ensure validation message is shown if the password is less than 8 characters upon submitting the form.
- The email field should autocomplete if possible.
- The current password field should autocomplete if possible.

Change password form:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with any empty fields.
- Ensure validation message is shown if the new password is less than 8 characters upon submitted the form.
- Ensure validation message is shown if the new password and confirmation password fields don't match upon submitting the form.
- The current password field should autocomplete if possible.
- The new password fields should autocomplete if possible.

Email form during sign in:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with an empty email field.
- Ensure a validation tool-tip is shown when submitting the form with an invalid email format.  The form should not allow the user to submit.

Password form during sign in:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with an empty password field.
- All existing validation messages are shown when the form is submitted
- The password field should autocomplete if possible.

Profile form:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with an empty required field.
- Ensure that you cannot submit the form with an empty "state" field if the country is "Unite States". 
- The date of birth field should autocomplete if possible.
- Company name field should autocomplete if possible.
- Title field should autocomplete if possible.
- All other validations should perform as expected.

Register email form:
- Validations are only displayed when the form is submitted.
- Ensure you cannot submit the form with an empty required field.
- Ensure a validation tool-tip is shown when submitting the form with an invalid email format.  The form should not allow the user to submit.

Register password form:
- New password field should autocomplete if possible.

Additional profile information modal:
- Validations are only displayed when the form is submitted. 
- Ensure you cannot submit the form with an empty Highest level of education field.
